keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -1261,6 +1261,26 @@ class SqlArtifactRepository(
     }
   }
 
+  override fun getArtifactPromotionStatus(
+    deliveryConfig: DeliveryConfig,
+    artifact: DeliveryArtifact,
+    version: String,
+    environmentName: String
+  ): PromotionStatus? {
+    return sqlRetry.withRetry(READ) {
+
+      jooq
+        .select(ENVIRONMENT_ARTIFACT_VERSIONS.PROMOTION_STATUS)
+        .from(ENVIRONMENT_ARTIFACT_VERSIONS)
+        .where(ENVIRONMENT_ARTIFACT_VERSIONS.ENVIRONMENT_UID.eq(deliveryConfig.getUidFor(environmentName)))
+        .and(ENVIRONMENT_ARTIFACT_VERSIONS.ARTIFACT_UID.eq(artifact.uid))
+        .and(ENVIRONMENT_ARTIFACT_VERSIONS.ARTIFACT_VERSION.eq(version))
+        .orderBy(ENVIRONMENT_ARTIFACT_VERSIONS.DEPLOYED_AT.desc())
```

Comment:
Why do you need to order by the deployed at time? I thought env + artifact + version should be unique in this table.

Reply:
You're right. This was copy/paste from another function, will remove.